subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/repositories/DefaultMavenArtifactRepository.java

```diff
@@ -139,11 +143,11 @@ public void setArtifactUrls(Iterable<?> urls) {
 
     @Override
     public void useGradleMetadata() {
-        preferGradleMetadata = true;
+        experimentalFeatures.enableAll();
```

Comment:
:o: Why do you use `enableAll()` here, and not specifically consume module metadata? Also this indicates that we have no longer control for each repository. This probably means that we will need an opt-out for repositories known to _not_ contain Gradle metadata (in order to avoid too many lookups in the future)

Reply:
Yes, we'll likely need a way to opt-out of module metadata lookups for performance reasons. We also plan to provide an opt-out for the current behaviour of checking for a `jar` file when the POM/Ivy file isn't found.

We agreed to think about 'experimental features' as something you enable as a whole, and not separate features to be enabled and disabled.

The control of separate named features is only possible via system properties at this stage, and should only be used to help us in debugging issues with these features.